### PR TITLE
Host navi static assets in elide

### PR DESCRIPTION
**Pull request by aklish**
#Description
1.  Builds Navi UI and hosts static assets in the Elide service.
2. Adds a controller to Elide service for Navi server-generated-config.js  

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/webservice/app/src/main/kotlin/com/yahoo/navi/ws/controllers/ServerGeneratedConfig.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class ServerGeneratedConfig @Autowired constructor(naviSettings: NaviConfig, elideSettings: ElideConfigProperties) {
+    private val mapper = ObjectMapper()
+    private val naviSettings: NaviConfig
+    private val elideSettings: ElideConfigProperties
+
+    @GetMapping(value = ["/assets/server-generated-config.js"], produces = ["application/javascript"])
+    @ResponseBody
+    @Throws(JsonProcessingException::class)
+    fun serverGeneratedConfig(user: Principal): String {
+        naviSettings.appSettings.user = user.name ?: ""
```
Comment (jkusa) [inline packages/webservice/app/src/main/kotlin/com/yahoo/navi/ws/controllers/ServerGeneratedConfig.kt:29]: empty ` user.name` is probably an error.

File: packages/webservice/app/src/main/kotlin/com/yahoo/navi/ws/controllers/ServerGeneratedConfig.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        if (naviSettings.appSettings.persistenceApiHost == "") {
```
Comment (jkusa) [inline packages/webservice/app/src/main/kotlin/com/yahoo/navi/ws/controllers/ServerGeneratedConfig.kt:31]:
```suggestion
        if (naviSettings.appSettings.persistenceApiHost.isBlank()) {
```

File: packages/webservice/app/src/test/kotlin/com/yahoo/navi/ws/test/integration/ServerGeneratedConfigTest.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    @BeforeEach
+    override fun setUp() {
+        super.setUp()
+        RestAssured.basePath = "/"
+    }
+
+    @Test
+    fun server_generated_config() {
+        var expected: String? =
```
Comment (jkusa) [inline packages/webservice/app/src/test/kotlin/com/yahoo/navi/ws/test/integration/ServerGeneratedConfigTest.kt:24]:
```suggestion
        val expected: String? =
```

File: packages/webservice/app/src/test/kotlin/com/yahoo/navi/ws/test/integration/StaticAssetServingTest.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        val content = this.javaClass::class.java.getResource("/META-INF/resources/index.html").readText()
+
+        given()
+            .header("User", "testuser")
+            .When()
+            .get("/")
+            .then()
+            .log().all()
+            .body(equalTo(content))
+            .statusCode(200)
```
Comment (jkusa) [inline packages/webservice/app/src/test/kotlin/com/yahoo/navi/ws/test/integration/StaticAssetServingTest.kt:33]: Can we assert the redirect url?

File: packages/webservice/app/src/main/kotlin/com/yahoo/navi/ws/controllers/ServerGeneratedConfig.kt (hunk 1 above)
Comment (aklish) [reply packages/webservice/app/src/main/kotlin/com/yahoo/navi/ws/controllers/ServerGeneratedConfig.kt:29]: User can't be null in this case.  Kotlin was actually giving a warning earlier about using the ? mark operator here.

File: packages/webservice/app/src/test/kotlin/com/yahoo/navi/ws/test/integration/StaticAssetServingTest.kt (hunk 4 above)
Comment (aklish) [reply packages/webservice/app/src/test/kotlin/com/yahoo/navi/ws/test/integration/StaticAssetServingTest.kt:33]: I tested that this was working manually with a local server.  I wasn't able to capture the redirect with Rest Assured.

**Review by cythrawll — APPROVED** [review]

**Pull request merged**